src/native/libs/System.Native/pal_io.c

```diff
@@ -369,6 +369,48 @@ int32_t SystemNative_Unlink(const char* path)
     return result;
 }
 
+int32_t SystemNative_MemfdSupported(void)
+{
+#if HAVE_MEMFD_CREATE
+#ifdef TARGET_LINUX
+    struct utsname uts;
+    int32_t major, minor;
+
+    // memfd_create is only known to work properly on kernel version > 3.17 and throws SIGSEGV instead of ENOTSUP
+    if (uname(&uts) == 0 && sscanf(uts.release, "%d.%d", &major, &minor) == 2 && (major < 3 || (major == 3 && minor < 17)))
+    {
+        return 0;
+    }
+#endif
+
+    int32_t fd = memfd_create("test", MFD_CLOEXEC);
```

Comment:
nit: when I saw it for the first time, I've asked myself if the name "test" is not common enough to cause troubles. Then I've read https://man7.org/linux/man-pages/man2/memfd_create.2.html and found the answer:

>  Names do not affect the behavior of the file descriptor, and as such multiple files can have the same name without any side effects.

IMO it would be nice to have a comment that explains it here.

Reply:
```suggestion
    // From linux manpage: 
    //   Names do not affect the behavior of the file descriptor, and as such multiple files can have the same name without any side effects.

    int32_t fd = memfd_create("test", MFD_CLOEXEC);
```